### ISSUE TITLE
Documentation for JointModelRevoluteUnbounded (why nq=2?)

### ISSUE THREAD

**Issue by stephane-caron** [stack-of-tasks/pinocchio#1751]
TL;DR: why do unbounded revolute joints have ``nq=2`` while bounded ones have ``nq=1``?

I'm running into an instance of https://github.com/stack-of-tasks/pinocchio/issues/65 for the [JointModelRevoluteUnbounded](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/structpinocchio_1_1JointModelRevoluteUnboundedTpl.html), which appears when a URDF has ``continuous`` joints.

* According to [urdf/XML/joint](https://wiki.ros.org/urdf/XML/joint), a ``continuous`` joint is "a continuous hinge joint that rotates around the axis and has no upper and lower limits". That's the same as ``revolute``, but without limits.
* In Pinocchio, it seems continuous joints have ``nq=2``, while revolute joints have ``nq=1``. Why is that so?

Here is for instance part of the [Kinova Gen2](https://github.com/Gepetto/example-robot-data/tree/master/robots/kinova_description) description from example-robot-data:

```
 JointModelRUBZ
   index: 4
   index q: 4
   index v: 3
   nq: 2
   nv: 1,
 JointModelRZ
   index: 5
   index q: 6
   index v: 4
   nq: 1
   nv: 1,
```

**Comment by wxmerkt** [comment]
The complex representation of SO(2) of the unbounded rotation is represented as `cos(theta), sin(theta)` - cf. https://github.com/stack-of-tasks/pinocchio/issues/794 - particularly https://github.com/stack-of-tasks/pinocchio/issues/794#issuecomment-490049010 and https://github.com/stack-of-tasks/pinocchio/issues/794#issuecomment-490050573
